### DISCUSSION TITLE
Primal solution failed for the baseline design

### DISCUSSION THREAD

**Discussion by byeonghoon-ahn** [mdolab/dafoam#196] [General]
Hello, I drew the grid with ICEM and imported it into dafoam. Then I changed the parameter like this,    
checkMeshThreshold": {"maxAspectRatio": 3.6e5, "maxNonOrth": 91.0, "maxSkewness": 8.0},
. After calculation, I found the following error.

![image](https://user-images.githubusercontent.com/77006538/139422112-8ee0fb94-1b5c-4cdf-9fa6-c258849b29e8.png)
 I want to ask what is the meaning of face pyramids: 1012 faces are incorrectly oriented in this error, and then how to solve it?, and then I set the parameters to this, will it affect the final optimization result?

**Comment by friedenhe** [comment]
This means some faces are flipped. I suggest you use snappyHexMesh, instead of ICEM, to generate the mesh for OpenFOAM. A good mesh in ICEM may be very bad for OpenFOAM; the ways they compute the mesh quality criteria are different.

**Reply by byeonghoon-ahn** [reply]
Thank you for your answer. I'll give it a try. I see that the previous example uses pyhyp to generate the grid. How about using ICEM generate surface grid and then using  pyhyp generate volumne mesh for OpenFOAM?
